### PR TITLE
Revert #7787

### PR DESCRIPTION
**Pull request by thorn0**
#7787 wasn't tested well enough with nested expressions. Also it's primary goal was to fix an issue with JSDoc type casts, but it was fixed by other changes.

cc @shisama

**Prettier pr-7787**
[Playground link](https://deploy-preview-7787--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEBDABAXgDpXegIwEsAzCAJygGFyioBzOcgWQgGcALOAEwAUmADgBs4ADwDKqAO7cm6ANS58ZCAEYFS9CoBMC9AHoAVOkgBbUwhjrD+zSoDMeoyYjnLum5oDWEMB1QEVBAQRHAA4qgWAPJQUQQQIgCqUFJwqABucE76Lm6w9pr0QkSmbOFCAJ4CHGwAEqgAXqjk3ABywQJsACqhPXAA3Li4xGSUNHSMLOxcfIIiEtKy5LgAPgB8WsGqqxs6BsZmFrDWtlDrmxCOzofu6J5nGz5+AUEh4ZFwMXEJcMmpGXAdugiiUymFKtU6o1mm0Ot1eqFBlAQAAaEAQAQwIjQNjIUDNcgQKS8ZoIXEoVBCKSoCq4tEEcioMBeOAwcQCJkTZAwcgAVzgaLgpgIPFk3AAMqgGLzUIwAGIUUyoGBYhjINC8mAQVEgDgwUxCADqHCI8DYHLAcHEZNNRHSpoq6rAbDpIDoZXIMF4jPoSuQJEpZTRACs2KIAEKM5msyQWcV0OD+wMCkChiQTEQARV5EHgSaEQZAHPIHvVBACcCEOoEtFghqI3BgHGQAA4AAxomvsOCGxkCdU1uAezI6gCOOfg3ox5LQbAAtFA4DweDryHBx0Q197ZX6kAGCymyqYiNy+YeM3Bs7nE3vk2iYAF643m0htPfGURigwguZUOqhwArDqvJlF0ATkvuhbpPyACSUCyLA4hgLQmIAILweIMAVCI+ZlAAvnhQA)
```sh
--parser babel
```

**Input:**
```jsx
a =
  bifornCringerMoshedPerplexSawder +
  foo1 +
  foo2 + /* comment1 */
  foo3 + /* comment2 */
  kochabCooieGameOnOboleUnweave + // comment3
  glimseGlyphsHazardNoopsTieTie;

bifornCringerMoshedPerplexSawder
|> foo1
|> foo2 /* comment1 */
|> foo3 /* comment2 */
|> kochabCooieGameOnOboleUnweave
|> glimseGlyphsHazardNoopsTieTie;

```

**Output:**
```jsx
a =
  bifornCringerMoshedPerplexSawder +
  foo1 +
  foo2 +
  /* comment1 */
  foo3 +
  /* comment2 */
  kochabCooieGameOnOboleUnweave + // comment3
  glimseGlyphsHazardNoopsTieTie;

bifornCringerMoshedPerplexSawder
  |> foo1
  |> foo2 |>
  /* comment1 */
  foo3 |>
  /* comment2 */
  kochabCooieGameOnOboleUnweave
  |> glimseGlyphsHazardNoopsTieTie;

```

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by fisker** [comment]
Sorry!! I click the wrong button , I was going to update another PR.

File: tests/js/babel-plugins/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -1138,6 +1138,30 @@ super.f(?); // partial application allowed for call on |SuperProperty|
 ================================================================================
 `;
 
+exports[`pipeline-operator-comments.js format 1`] = `
```
Comment (fisker) [inline tests/js/babel-plugins/__snapshots__/jsfmt.spec.js.snap:1141]: Should put in `comments-pipeline-own-line` or `pipeline-operator` dir.

**Comment by fisker** [comment]
We should open an issue, and discuss the following two cases, after we merge this one.

**Prettier pr-7787**
[Playground link](https://deploy-preview-7787--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEAjAlgMwgJygYR3SgHM4cBZCAZwAs4ATABXIAcAbOADwGUBDAO4NyAAhEAdKGIA+APhEB6AFQjIAWzUIYAJhFKFksdggSpIuYpXrNsAMx6DZgNYQwtPqnwQI6OAHE+TQB5KCDUCE4AVSgBOD4ANzgxQ3NZFJJ2dDVqf3YAT1ZaagAJPgAvPhwGADlvVmoAFV8muABuSUkMbDxCYjJKGnpmNk5eQWEcZLMLBQVVCA0tFKNvFJm56yWzMRc3Dy8ff0C4ELCIuGjYhKS1tO2RDKycv3zCkvLKmrrG5t9JEAANCAIKwYOhoNRkKBKjgIAImJUEJCUHx2AI+HlIUDUDg+GAnHAYDxWHi+sgYDgAK5wIFwNSoRjCBgAGT4pEpfDIADFcGo+DAwaRkCA+JSYBBASBaDA1OwAOq0dDwagksBwHhIpXoeJKvLCsDULEgYg5HAwJi4kh85CYVE5IEAK2oXAAQrj8YT+JpmcQ4Da7TSQE7eH1OABFSkQeD+9j2kAknCm4WoDxwdiS1hEWBy9AMGC0ZAADgADEDMzQ4HLcaxhZm4KbEpKAI6R+AWkHIkXUAC0UDgjEYkpwcBb6GHFs51qQttjgZyanQ5Kpc9DcAjUb904DQJgHhzeYLSG0O9x6EypC8Gj4wvrAFZJZScg0PMiZ3H4tSAJJQYSwHhgIhQQAQR-HgYDyTgYxyABfaCgA)
```sh
--parser babel
```

**Input:**
```jsx
bifornCringerMoshedPerplexSawder  
  |> /* comment2 */
  foo 
  |> /* comment3 */
  kochabCooieGameOnOboleUnweave  
  |>
  glimseGlyphsHazardNoopsTieTie;

bifornCringerMoshedPerplexSawder  
  |> // comment
    foo
  |> // comment
    kochabCooieGameOnOboleUnweave 
  |>
    glimseGlyphsHazardNoopsTieTie

```

**Output:**
```jsx
bifornCringerMoshedPerplexSawder |>
  /* comment2 */
  foo |>
  /* comment3 */
  kochabCooieGameOnOboleUnweave
  |> glimseGlyphsHazardNoopsTieTie;

bifornCringerMoshedPerplexSawder  // comment
|> foo  // comment
  |> kochabCooieGameOnOboleUnweave
  |> glimseGlyphsHazardNoopsTieTie;

```

**Comment by thorn0** [comment]
Hopefully they will be fixed by #8491.

**Comment by thorn0** [comment]
@fisker Confirmed. #8491 fixes those cases.

**Pull request merged**